webhook/webhook_test.go

```diff
@@ -32,20 +32,30 @@ import (
 )
 
 func newDefaultOptions() Options {
+	statsReporter, err := NewStatsReporter()
```

Comment:
We need to make sure that web_integration_test suite works without passing a StatsReporter as previously. Right now if I comment out this part we have a race condition. I strongly recommend to move the change 
```
if opts.StatsReporter == nil {
		RegisterMetrics()
```
to a new PR.

Reply:
I removed those changes as you requested, but allow me explaining why I included them in the first place:

knative/pkg consumers still won't be able to use a customized `StatsReporter` that excludes certain tags, as double metrics registration won't work.

I can send a followup to address the data race.

Reply:
@zhouhaibing089 hi,
I am not sure about what is the use case we are targeting but I see two things to cover:

a) default reporter with excluded tags
b) custom reporter with excluded tags

Could we just have (main):
```
	if len(webhooks) > 0 {
		opts := GetOptions(ctx)
		// Register webhook metrics
		webhook.RegisterMetrics(opts...)
```
so with this PR to support a) and then later decide about b)?  wdyth?

Reply:
Sorry for my late response.

I'm looking for a way to exclude tags, either "a" or "b" would be fine to me, and "a" seems to be more appealing. I think what you suggested definitely makes it easier for downstream. Let me try that way.

(Sorry I tended to amend my commits when there are comments that need to be addressed, I'll try append going forwards in this repo.)

Reply:
I pushed a second commit to add `StatsReporterOptions` in `webhook.Options`.